gadget/update.go

```diff
@@ -301,6 +305,43 @@ func onDiskStructureIsLikelyImplicitSystemDataRole(gadgetVolume *Volume, diskLay
 		numPartsInGadget+1 == numPartsOnDisk
 }
 
+func ensureVolumeEMMCCompatibility(gadgetVolume *Volume, diskVolume *OnDiskVolume) (map[int]*OnDiskStructure, error) {
+	gadgetStructIdxToOnDiskStruct := map[int]*OnDiskStructure{}
+	for _, gs := range gadgetVolume.Structure {
+		// ensure the device node exists
+		emmcNode := fmt.Sprintf("%s%s", diskVolume.Device, gs.Name)
```

Comment:
Wouldn't it be better to check the directory in `/sys/block/`?

Reply:
You could have something that is not really related to the volume we are looking at I guess, as we are just looking by name. Probably there is a way to know for sure if a given device has special emmc partitions. Maybe not worth doing it, but at least a TODO would be good to have.

Reply:
added a TODO!